xc/device.ml

```diff
@@ -1244,8 +1244,10 @@ let clean_shutdown (task: Xenops_task.t) ~xs (x: device) =
 			try (Xenctrl.domain_getinfo xc x.frontend.domid).Xenctrl.hvm_guest
 			with _ -> false
 			in
-		try release ~xc ~xs ~hvm devs x.frontend.domid x.frontend.devid
-		with _ -> ());
+		if hvm then 
```

Comment:
For _hot_ unplug (i.e. while the guest is running), that is true, but I think that we may still need to clean up the PCI state after a shutdown by calling pci-detach. The existing code seems to call this for PV in this function, and I'm worried about breaking something if we stop doing it. The main motivation for this patch was to use `xl pci-{attach,detach}` for HVM, while it was already used for PV, so I think we should change the PV code paths (at least, not as part of this patch).